shared/src/permissions.rs

```diff
@@ -0,0 +1,157 @@
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use strum::IntoEnumIterator;
+
+#[derive(Clone, Copy, Hash, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub enum Permission {
+    Account(AccountPermission),
+    Universe(GenericPermission),
+}
+
+#[derive(
+    Clone,
+    Copy,
+    Hash,
+    Eq,
+    PartialEq,
+    Debug,
+    Serialize,
+    Deserialize,
+    strum_macros::EnumString,
+    strum_macros::ToString,
+    strum_macros::EnumIter,
+)]
+pub enum AccountPermission {
+    PermanentBan,
+    TemporaryBan,
+    List,
+    View,
+}
+
+impl ToString for Permission {
```

Comment:
Rust recommends to never implement `ToString`, but to implement `Display` instead, which implements `ToString` automatically.